blocks/base.py

```diff
@@ -87,13 +87,19 @@ def deplate(
         plugins: Optional[List["Plugin"]] = None,
         train: Optional[bool] = None,
     ) -> pd.Series:
-        if not hasattr(self, "data") and hasattr(self, "category"):
-            self.data = self.dataloader.load(self.category)
-
         return self.data[Const.input_col]
 
-    def load(self) -> None:
-        pass
+    def load(self, plugins: List["Plugin"]) -> None:
+        if not hasattr(self, "data") and hasattr(self, "category"):
+            self.data = self.dataloader.load(self.category)
+            logger.log(
+                f"✅ Loaded dataloader on datasource: {self.id}", level=logger.levels.ONE
+            )
+        else:
+            logger.log(
+                f"❌ Couldn't load dataloader on datasource: {self.id}",
```

Comment:
can we throw an exception?

Reply:
👍

Reply:
I think we sold move this to deplate, so we only do the loading at the time and place when we need it.

Reply:
It's at the time and place where we need it if you think about it. A pipeline has either one or two datasources that get called straight after ``load`` either in ``train`` or in ``predict``